sdk/core/Microsoft.Extensions.Azure/src/Internal/ClientOptionsFactory.cs

```diff
@@ -36,27 +36,53 @@ private TOptions CreateOptions(string name)
             }
 
             ConstructorInfo parameterlessConstructor = null;
-            ParameterInfo versionConstructor = null;
+            int versionParameterIndex = 0;
```

Comment:
Do you need to support this?  https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-versioning-select-api-version says the version always has to come first.